xarray/core/formatting_html.py

```diff
@@ -125,6 +125,40 @@ def summarize_vars(variables):
     return f"<ul class='xr-var-list'>{vars_li}</ul>"
 
 
+def short_index_repr_html(index):
+    if hasattr(index, "_repr_html_"):
+        return index._repr_html_()
+
+    return f"<pre>{escape(repr(index))}</pre>"
+
+
+def summarize_index(coord_names, index):
+    name = "<br>".join([escape(str(n)) for n in coord_names])
+
+    index_id = f"index-{uuid.uuid4()}"
+    preview = escape(inline_index_repr(index))
+    details = short_index_repr_html(index)
+
+    data_icon = _icon("icon-database")
+
+    return (
+        f"<div class='xr-index-name'><div>{name}</div></div>"
+        f"<div class='xr-index-preview xr-preview'>{preview}</div>"
```

Comment:
For slightly better contrast, you can remove the `xr-preview` class here

```suggestion
        f"<div class='xr-index-preview'>{preview}</div>"
```

and update CSS with 

```css
.xr-index-preview {
  grid-column: 2 / 5;
  color: var(--xr-font-color2);
}
```